.circleci/create_circleci_config.py

```diff
@@ -101,16 +106,21 @@ def to_dict(self):
             {
                 "restore_cache": {
                     "keys": [
-                        f"v{self.cache_version}-{self.cache_name}-pip-" + '{{ checksum "setup.py" }}',
-                        f"v{self.cache_version}-{self.cache_name}-pip-",
+                        # check the fully-matched cache first
+                        f"v{self.cache_version}-{self.cache_name}-{cache_branch_prefix}-pip-" + '{{ checksum "setup.py" }}',
+                        # try the partially-matched cache from `main`
+                        f"v{self.cache_version}-{self.cache_name}-main-pip-",
```

Comment:
Similar to `main` vs `pull`, we don't want `pr_2`  being affected by `pr_1`.

We can be confident to use the cache used in `main`:  the problem happens only **when we use dev. version on `main` (say accelerate), and we decide to use the un-pinned release version in a PR (that is behind the pinned dev. version).**